holidays/countries/bahrain.py

```diff
@@ -0,0 +1,177 @@
+        # in some instances prepending/appending a day or two
+        # before/after the official holiday.
+        fitr = "Eid Al Fitr"
+        if year in dates_obs:
+            for date_obs in dates_obs[year]:
+                hol_date = date(year, *date_obs)
+                self[hol_date] = fitr
+                self[hol_date + rd(days=1)] = f"{fitr} Holiday"
+                self[hol_date + rd(days=2)] = f"{fitr} Holiday"
+        else:
+            for yr in (year - 1, year):
+                for date_obs in _islamic_to_gre(yr, 10, 1):
+                    hol_date = date_obs
+                    _add_holiday(hol_date, f"{fitr}* (*estimated)")
+                    _add_holiday(
+                        hol_date + rd(days=1),
+                        f"{fitr} Holiday* (*estimated)",
+                    )
+                    _add_holiday(
+                        hol_date + rd(days=2),
+                        f"{fitr} Holiday* (*estimated)",
+                    )
+
+        # Eid Al Adha
+        # Date is announced each year. Usually stretches along 3 or 4 days,
+        # in some instances prepending/appending a day or two
+        # before/after the official holiday.
+        adha = "Eid Al Adha"
+        if year in dates_obs:
+            for date_obs in dates_obs[year]:
+                hol_date = date(year, *date_obs)
+                self[hol_date + rd(days=1)] = adha
+                self[hol_date + rd(days=2)] = f"{adha} Holiday"
+                self[hol_date + rd(days=3)] = f"{adha} Holiday"
+        else:
+            for yr in (year - 1, year):
+                for date_obs in _islamic_to_gre(yr, 12, 9):
+                    hol_date = date_obs
+                    _add_holiday(
+                        hol_date + rd(days=1), f"{adha}* (*estimated)"
+                    )
+                    _add_holiday(
+                        hol_date + rd(days=2),
+                        f"{adha}* Holiday* (*estimated)",
+                    )
+                    _add_holiday(
+                        hol_date + rd(days=3),
+                        f"{adha} Holiday* (*estimated)",
+                    )
+
+        # Al Hijra New Year
+        new_hijri_year = "Al Hijra New Year"
+        if year in dates_obs:
+            for date_obs in dates_obs[year]:
+                hol_date = date(year, *date_obs)
+                self[hol_date] = new_hijri_year
+        else:
+            for date_obs in _islamic_to_gre(year, 1, 1):
+                hol_date = date_obs
+                self[hol_date] = f"{new_hijri_year}* (*estimated)"
+
+        # Ashoora
+        # Date is announced each year, for the 9th and 10th Day,
+        # of the month of Muharam
+        ashoora = "Ashoora"
+        if year in dates_obs:
+            for date_obs in dates_obs[year]:
+                hol_date = date(year, *date_obs)
+                self[hol_date + rd(days=9)] = ashoora
+                self[hol_date + rd(days=10)] = f"{ashoora} Holiday"
```

Comment:
If Ashoora calculated as 9 Muharram, it's no need to add 9
```suggestion
                self[hol_date] = ashoora
                self[hol_date + rd(days=+1)] = f"{ashoora} Holiday"
```